test/ServiceBus.Tests/PlugableQueueBalancerTests/PluggableQueueBalancerTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using Orleans.ServiceBus.Providers.Testing;
+using Orleans.Storage;
+using Orleans.Streams;
+using Orleans.TestingHost;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TestExtensions;
+using Xunit;
+
+namespace Tester.StreamingTests
+{
+    [TestCategory("EventHub"), TestCategory("Streaming")]
+    public class PluggableQueueBalancerTests : OrleansTestingBase, IClassFixture<PluggableQueueBalancerTests.Fixture>
+    {
+        private static string QueueBalancerFactoryName = "CustomBalancerFactory";
+        private const string StreamProviderName = "EventHubStreamProvider";
+        private const string StreamNamespace = "EHTestsNamespace";
+        private static readonly TimeSpan timeout = TimeSpan.FromSeconds(30);
+        private static readonly int totalQueueCount = 6;
+        private static readonly short siloCount = 2;
+        public static readonly EventHubGeneratorStreamProviderSettings ProviderSettings =
+            new EventHubGeneratorStreamProviderSettings(StreamProviderName);
+        public static readonly PersistentStreamProviderConfig PersistentProviderConfig =
+            new PersistentStreamProviderConfig();
+
+        private readonly Fixture fixture;
+
+        public class Fixture : BaseTestClusterFixture
+        {
+            protected override TestCluster CreateTestCluster()
+            {
+                var options = new TestClusterOptions(siloCount);
+                ProviderSettings.EventHubPartitionCount = totalQueueCount;
```

Comment:
Would prefer if these tests where in a test harness so we could run them against other persistent providers.
Initial test should probably use memory stream provider, It allows the same queue flexibility, while also allowing us to do this in a sharable harness.